dag.go

```diff
@@ -65,6 +66,63 @@ func TopologicalSort[K comparable, T any](g Graph[K, T]) ([]K, error) {
 	return order, nil
 }
 
+func StableTopologicalSort[K comparable, T any](g Graph[K, T], less func(K, K) bool) ([]K, error) {
+	if !g.Traits().IsDirected {
+		return nil, fmt.Errorf("topological sort cannot be computed on undirected graph")
+	}
+
+	predecessorMap, err := g.PredecessorMap()
+	if err != nil {
+		return nil, fmt.Errorf("failed to get predecessor map: %w", err)
+	}
+
+	queue := make([]K, 0)
+
+	for vertex, predecessors := range predecessorMap {
+		if len(predecessors) == 0 {
+			queue = append(queue, vertex)
+		}
+	}
+
+	order := make([]K, 0, len(predecessorMap))
+	visited := make(map[K]struct{})
+
+	for len(queue) > 0 {
+		sort.Slice(queue, func(i, j int) bool {
+			return less(queue[i], queue[j])
+		})
+
```

Comment:
The implementation is the same as `TopologicalSort`, but `StableTopologicalSort` additionally sorts the vertices currently in the queue using the provided `less` function in each iteration.